README
/**************************************************
  CPG MiniCMS Plugin for Coppermine Photo Gallery
  *************************************************
  CPGMiniCMS version: 1.0 - 1.6
  Copyright (c) 2005-2006 Donovan Bray <donnoman@donovanbray.com>
  *************************************************
  1.3.0  eXtended miniCMS
  Copyright (C) 2004 Michael Trojacher <m.trojacher@webtips.at>
  Original miniCMS Code (c) 2004 by Tarique Sani <tarique@sanisoft.com>,
  Amit Badkas <amit@sanisoft.com>
  *************************************************
  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.
  *************************************************
  Coppermine version: 1.4.9
  CPGMiniCMS version: 1.7A1
  $Source: /cvsroot/cpg-contrib/minicms/README,v $
  $Revision: 1.6 $
  $Author: donnoman $
  $Date: 2006/11/10 21:26:26 $
***************************************************/

What it is
----------

A minimalistic Content Mangement System which allows the gallery admin to
add / edit / delete pages. Typically this will be used by people who just want
to add a few pages like About, Contact etc to their installs of coppermine but
do not want full fledged CMS like Nuke.

It is also possible to include a category info page, which can be shown when a
user visits a specific category in your gallery.

It has FCKEditor WYSIWYG editor which works in IE and Mozilla, and respects the
admin rights from Coppermine.


INSTALL
-------

* Unpack the archive and upload the "minicms" directory structure to your Coppermine
  gallery's plugins folder.
* Coppermine should be 1.4.9 or above to work with CPG MiniCMS 1.6.
* Login as an admin, go to config, then "Manage Plugins"
* Find the CPG MiniCMS entry in the available plugins and click install.
* You can adjust the placment on the main page by moving the token named "minicms"
  in the "Content of the Main Page" field in the config.
* When logged in as admin you will have an option to edit the content block if it
  already has content.
* When logged in as admin you will be shown an empty block with an edit button when
  the category has no content.
* To review your MiniCMS click on the "MiniCMS Admin" button on the "admin menu".
* To change your MiniCMS configuration use the "MiniCMS Config" button on the
  "Plugin Manager" page.
* You can include any article in its own page to be used as a hyperlink by using
  it's ID i.e.: http://www.donovanbray.com/index.php?file=minicms/cms&id=13


CHANGELOG
/**************************************************
  CPG MiniCMS Plugin 1.7 for Coppermine Photo Gallery 1.4.9
  *************************************************
  CPGMiniCMS version: 1.0 - 1.7
  Copyright (c) 2005-2006 Donovan Bray <donnoman@donovanbray.com>
  *************************************************
  1.3.0  eXtended miniCMS
  Copyright (C) 2004 Michael Trojacher <m.trojacher@webtips.at>
  *************************************************
  Original miniCMS Code (c) 2004 by Tarique Sani <tarique@sanisoft.com>,
  Amit Badkas <amit@sanisoft.com>
  *************************************************
  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.
  *************************************************
  $Source: /cvsroot/cpg-contrib/minicms/CHANGELOG,v $
  $Revision: 1.19 $
  $Author: donnoman $
  $Date: 2007/02/10 21:24:15 $
***************************************************/
RELEASED RC2
ADDED Display of SQL entries in the configure section on plugin install.
CHANGED pos column name in _cms table to cpos to avoid conflicts with Coppermine.
ADDED on plugin install the database updates are shown in pluginmgr.php.
RELEASED 1.3
ADDED plugin actions to put content below the breadcrumb on thumbnail.php and displayimage.php
FIXED Positioning code
Add "section" creation to the UI - Donnoman 3-12-2005
RELEASED 1.4
Added "related" page to display content that share keywords - TranzNDance 3-19-2005
Added themeable functions and templates - Donnoman 3-19-2005
Added fckeditor and tinymce - Donnoman 3-19-2005
Made fckeditor the default - Donnoman 3-19-2005
Added config page - Donnoman 3-20-2005
Added auto-configuration on db version change - Donnoman 3-20-2005
Fixed action buttons in codebase - TranzNDance 3-23-2005
Added config option to select active editor - Donnoman 4-2-2005
Added style.xml support to fckeditor and began using Coppermine's CSS in the editor - Donnoman 4-2-2005
Fixed problem where some Coppermine themes were using an id of "content", changed minicms to "minicms_content", Donnoman 4-10-2005
Added a link to comments for each image on "related" page - TranzNDance 9-27-2005
Added RSS capability on "related" page - TranzNDance 10-02-2005
Removed htmlarea and tinymce support files, they will be supported as add-ons - Donnoman 12-03-2005
RELEASED 1.5
Upgraded to FCKEditor 2.3.1
Misc Fixes
Moved configuration button from the admin toolbar to the plugin installed_info section on the plugin manager page.
RELEASED 1.6
Fixed issue where aborting new content creation left an orhpaned entry in the database titled "New Content".
Moved the preview to be above the editor, on the editor page.
Eliminated automatic preview.
Eliminated the unbound minicms edit box when veiwing a specific article.
Added the real content title of the first cms article to the html header on the cms page.
Fixed Issue with Bridged Installations
Changed it so that new MiniCMS articles aren't created until they are saved.
RELEASED 1.7




lang/english.php
<?php
/**************************************************
  CPG MiniCMS Plugin for Coppermine Photo Gallery
  *************************************************
  CPGMiniCMS version: 1.0 - 1.6
  Copyright (c) 2005-2006 Donovan Bray <donnoman@donovanbray.com>
  *************************************************
  1.3.0  eXtended miniCMS
  Copyright (C) 2004 Michael Trojacher <m.trojacher@webtips.at>
  Original miniCMS Code (c) 2004 by Tarique Sani <tarique@sanisoft.com>,
  Amit Badkas <amit@sanisoft.com>
  *************************************************
  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.
  *************************************************
  Coppermine version: 1.4.9
  CPGMiniCMS version: 1.7A1
  $Source: /cvsroot/cpg-contrib/minicms/lang/english.php,v $
  $Revision: 1.13 $
  $Author: donnoman $
  $Date: 2006/11/10 21:24:24 $
***************************************************/

/**
 * English Language File
 * @author Donovan Bray <donnoman@donovanbray.com>
 * @package language
 * @subpackage english
 * @version 1.7
 */

if (!defined('IN_COPPERMINE')) { die('Not in Coppermine...');}

$lang_minicms = array(
  'minicms' => 'MiniCMS', // Display Name
  'minicms_full' => 'cpg-contrib MiniCMS',
  'admin_title' => 'MiniCMS Admin', // Title of the button on the gallery admin menu
  'config_title' => 'MiniCMS Config', // Title of the button on the gallery admin menu
  'title' => 'Title', // Title of new content field on cms_admin form
  'page_success' => 'Page saved successfully', // Success messages
  'page_fail' => 'Error saving page', // Fail Messages
  'non_exist' => 'Content doesn\'t exist', // Error Message
  'action' => 'Action', // Column Heading
  'view' => 'View', // Column Heading
  'content' => 'Content', // Column Heading
  'edit' => 'Edit', // Column Heading
  'cpos' => 'Position', //Column Heading, renamed cpos to avoid conflict with coppermines pos field.
  'add_new' => 'Add new content', //Form Field
  'new_content' => 'New Content', // Default Title on new content from cms_admin
  'delete' => 'Delete', // Action
  'pos_up' => 'Raise Position', // Action
  'pos_down' => 'Lower Position', // Action
  'type' => 'Type', // Column Heading
  'article' => 'Article', // Column Heading
  'add' => 'Add', //Form Field
  'preview' => 'Preview', //Form Field
  'submit' => 'Submit', //Form Field
  'no_change' => 'No change required', //Error message
  'expand_all' => 'Expand All', // config
  'dbver_nomatch' => 'Current database version doesn\'t match',
  'comments_lnk' => 'Comments',
  'rss_more_lnk' => 'article',
  'no_rss' => 'RSS not available',
);

$lang_minicms_config_related_size =  array(
      'thumb'=>'thumb',
      'normal'=>'normal',
);

$lang_minicms_config_editor = array(
      'fckeditor' => 'fckeditor',
//      'htmlarea' => 'htmlarea',  //unremark if you've loaded the support files
//      'tinymce' => 'tinymce',    //unremark if you've loaded the support files
);

$lang_minicms_config = array(
// array('Short Descripiton','field name',field type,'Link to documentation'),
  'General Settings',
  array('Index Redirect Page', 'redirect_index_php', 0),
  array('Size of the images on the related page', 'related_size', 3,'',$lang_minicms_config_related_size),
  array('HTML WYSIWG Content Editor','editor',3,'',$lang_minicms_config_editor),
  'RSS',
  array('Enable RSS feed','rss_enabled',1,'',''),
  array('RSS description length','rss_description_length',0),
  array('Display image in feed','rss_include_image',1,'',''),
  array('Image size in feed','rss_image_size',3,'',$lang_minicms_config_related_size),
);



?>